File: 1.cpp
#include <fstream>
#include <iostream>
#include <string>

using namespace std;

int main() {
  ifstream in("input.txt");
  if (!in) {
    cerr << "Failed to open input.txt" << endl;
  }

  string line;
  int dial = 50;
  int p1 = 0;
  int p2 = 0;

  while (in >> line) {
    char d = line[0];
    int amt = stoi(line.substr(1));

    for (int i = 0; i < amt; i++) {
      if (d == 'L') {
        dial = (dial - 1 + 100) % 100;
      } else {
        dial = (dial + 1) % 100;
      }
      if (dial == 0) {
        p2++;
      }
    }
    if (dial == 0) {
          p1++;
    }
  }

  cout << "part 1 password: " << p1 << endl;
  cout << "part 2 password: " << p2 << endl;

  return 0;
}

File: 1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanWords)

	dial := 50
	p1 := 0
	p2 := 0

	for scanner.Scan() {
		line := scanner.Text()

		d := line[0]
		amtStr := line[1:]
		amt, err := strconv.Atoi(amtStr)
		if err != nil {
			continue
		}

		for i := 0; i < amt; i++ {
			if d == 'L' {
				dial = (dial - 1 + 100) % 100
			} else {
				dial = (dial + 1) % 100
			}
			if dial == 0 {
				p2++
			}
		}
		if dial == 0 {
			p1++
		}
	}

	fmt.Printf("part 1 password: %d\n", p1)
	fmt.Printf("part 2 password: %d\n", p2)
}
